common/src/main/java/org/openmrs/analytics/FhirStoreUtil.java

```diff
@@ -89,27 +89,36 @@ public void uploadResourceToCloud(String resourceType, String resourceId, Resour
 	}
 	
 	// TODO: merge the two versions of this method to remove redundant resource info
-	public void uploadResourceToCloud(String resourceType, String resourceId, String fhirJson) {
-		uploadResourceToCloud(resourceType, resourceId, (Resource) fhirContext.newJsonParser().parseResource(fhirJson));
+	public void uploadResourceToCloud(String resourceType, String resourceId, String fhirJson) throws Exception {
+		if (httpUtil.checkServerConnection("http://google.com")) {
+			uploadResourceToCloud(resourceType, resourceId,
+					(Resource) fhirContext.newJsonParser().parseResource(fhirJson));
+		} else {
+			log.error("please Check your internet Connectivity");
+		}
 	}
 	
 	private void updateFhirResource(String fhirStoreName, String resourceId, String resourceType, Resource resource)
-	        throws IOException, URISyntaxException {
+			throws IOException, URISyntaxException ,Exception{
 		// Initialize the client, which will be used to interact with the service.
 		CloudHealthcare client = createClient();
 		String uri = String.format("%sv1/%s/fhir/%s/%s", client.getRootUrl(), fhirStoreName, resourceType, resourceId);
 		URIBuilder uriBuilder = new URIBuilder(uri);
 		log.info(String.format("Full URL is: %s", uriBuilder.build()));
-		
+
 		StringEntity requestEntity = new StringEntity(fhirContext.newJsonParser().encodeResourceToString(resource),
-		        StandardCharsets.UTF_8);
-		
+				StandardCharsets.UTF_8);
+
 		HttpUriRequest request = RequestBuilder.put().setUri(uriBuilder.build()).setEntity(requestEntity)
-		        .addHeader("Content-Type", "application/fhir+json").addHeader("Accept-Charset", "utf-8")
-		        .addHeader("Accept", "application/fhir+json").addHeader("Authorization", "Bearer " + getAccessToken())
-		        .build();
-		String response = httpUtil.executeRequest(request);
-		log.debug("Update FHIR resource response: " + response);
+				.addHeader("Content-Type", "application/fhir+json").addHeader("Accept-Charset", "utf-8")
+				.addHeader("Accept", "application/fhir+json").addHeader("Authorization", "Bearer " + getAccessToken())
+				.build();
+		if (httpUtil.checkServerConnection(uriBuilder.build().toString())) {
```

Comment:
So here is my suggestion: Instead of relying on a separate `checkServerConnection`, implement a retry logic inside `HttpUtil.executeRequest` such that depending on a configurable parameter, it waits a few seconds and retries. If the problem persists, throw the exception in `executeRequest` and do not catch it elsewhere.